### PR TITLE
Bill's 03 homework for 9-17

### PR DESCRIPTION
**Pull request by queensburybill**
Submitting 03 homework assignment

### CODE REVIEW

File: exercises/03-foreach/03-foreach.js [hunk 1]
```diff
@@ -4,7 +4,26 @@
...
+Write a function called doubleValues which accepts an array and returns a new array with all the
+values in the array passed to the function doubled
+*
+Examples:
+doubleValues([1,2,3]) // [2,4,6]
+doubleValues([1,-2,-3]) // [2,-4,-6]
+*/
+function doubleValues(arr) {
+  let newArr = [];
+  arr.forEach(n => newArr.push(n * 2));
```
Comment (Louis345) [inline exercises/03-foreach/03-foreach.js]: Nice use of the double arrows. We will eventually be covering ES6 syntax, but good to know that you're familiar with it.

**Comment by queensburybill** [comment]
I finished all the 04 assignments but not Guessing Game. I also think I found a couple of errors in the tests. I revised the tests and my exercises passed. You can see the revisions in the pull request, but I list them here too: (1) Map Test - extractKey function is missing the second parameter. (2) Reduce Test - arrayToObject function had a loose object without a key or closing brackets.

File: exercises/03-currying/03-guessing-game.html [hunk 2]
```diff
@@ -58,6 +58,29 @@
              * @param {number} numberOfTries the number of guesses the user gets before they lose
              */
             function GuessingGame(numberOfTries) {
+            // DON'T LOOK HERE!!!!!!
```
Comment (Louis345) [inline exercises/03-currying/03-guessing-game.html]: I won't look I promise.

File: exercises/04-anagram/04-anagram.js [hunk 3]
```diff
@@ -14,7 +14,10 @@
  * @param {string} string2 second string to compare
  * @returns {boolean} whether or not the strings are anagrams
  */
-function anagrams() {}
+function anagrams(str1, str2) {
```
Comment (Louis345) [inline exercises/04-anagram/04-anagram.js]: A really nice job here. I like how the code is compact and easily readable

File: test/04-map/04-map-test.js [hunk 4]
```diff
@@ -33,6 +33,6 @@ it("tests extractedName to see if the function returns a new array with the valu
     { name: "Tim" },
     { name: "Matt" },
     { name: "Colt" }
-  ]);
+  ], 'name');
   expect(extractedName).to.be.equalTo(["Elie", "Tim", "Matt", "Colt"]);
```
Comment (Louis345) [inline test/04-map/04-map-test.js]: Great catch here. I must have forgotten to pass in the key there. Good job in looking at the test and figuring it out.

**Pull request closed without merging**